client/src/features/bulletins/Bulletins/Bulletins.tsx

```diff
@@ -8,37 +8,102 @@ import { selectBulletins } from '../../../store/reducers';
 import { BulletinPreview } from '../BulletinPreview/BulletinPreview';
 import { Link } from 'react-router-dom';
 import { PostLoader } from '../../ui/PostLoader/PostLoader';
+import { Filters } from './Filters/Filters';
 
 type Props = {
   history: History;
 };
 
+export enum hasFees {
+  YES = 'yes',
+  NO = '',
+  ANY = 'any',
+}
+
 export const Bulletins = (_: Props) => {
   const bulletins = useSelector(selectBulletins);
+  const [filteredBulletins, setFilteredBulletins] = useState(bulletins);
+
+  const [filters, setFilters] = useState({
+    minPrice: 0,
+    maxPrice: 1000,
+    fees: hasFees.ANY,
+  });
+
   const dispatch = useDispatch();
 
   useEffect(() => {
     dispatch(getBulletins());
   }, [dispatch]);
 
-  if (!bulletins) return <PostLoader />;
-  if (!bulletins.length)
+  const handleSetFilters = useCallback((minPrice: number, maxPrice: number, fees: hasFees) => {
+    setFilters({ minPrice, maxPrice, fees });
+  }, []);
+
+  useEffect(() => {
```

Comment:
Ho l'impressione che tu non abbia capito né `useEffect` né l'approccio di React in generale. In React si ricalcolano i dati al render in base allo stato e props, `useEffect` serve per fare side-effects dopo ogni render, ma calcolare la lista dei bulletins filtrati non è un side-effect, è uno stato derivato che vuoi ricalcolare ad ogni render ed è deterministico a parità di props e state. Al limite se vuoi evitare di farlo per questioni di performance usi `React.memo` ma non cambia il concetto di base